tests/unit/src/test/scala/tests/codeactions/ConvertToNamedArgumentsSuite.scala

```diff
@@ -0,0 +1,55 @@
+package tests.codeactions
+
+import scala.meta.internal.metals.codeactions.ConvertToNamedArguments
+import scala.meta.internal.metals.codeactions.ExtractValueCodeAction
+
+class ConvertToNamedArgumentsSuite
+    extends BaseCodeActionLspSuite("convertToNamedArguments") {
+
+  check(
+    "convert-to-named-args-basic",
+    """|object Something {
+       |  case class Foo(param1: Int, param2: Int, param3: Int)
+       |  Foo<<(>>1, 2, param3 = 3)
+       |}""".stripMargin,
+    s"""|${ConvertToNamedArguments.title}
+        |""".stripMargin,
+    """|object Something {
+       |  case class Foo(param1: Int, param2: Int, param3: Int)
+       |  Foo(param1 = 1, param2 = 2, param3 = 3)
+       |}""".stripMargin,
+    selectedActionIndex = 0
+  )
+
```

Comment:
We need more test cases for example:

- `def` case (in addition to the case class's apply method)
- `Foo<<(>>1, param2 = 2, 3)` and `Foo<<(>>param1 = 1, 2, 3)`
- if there're multiple paramss like `foo(x: Int, y: Int)(z: Int)`

Reply:
In addition, it would be awesome if we have a test using external libraries

you can create a test using scala standard libraries by 

```scala
class ConvertToNamedArgumentsSuite extends BaseCodeActionSuite {
   override def requiresScalaLibrarySources: Boolean = true
```

in `tests.pc`, refer 
- https://github.com/scalameta/metals/blob/0f6851d77d673368382019be6db9842b11f8b643/tests/cross/src/test/scala/tests/pc/AutoImplementAbstractMembersSuite.scala 
- https://github.com/scalameta/metals/blob/0f6851d77d673368382019be6db9842b11f8b643/tests/cross/src/test/scala/tests/pc/InsertInferredTypeSuite.scala

Reply:
Thanks! I added more tests and a [new suite](https://github.com/camgraff/metals/blob/feat/convert_to_named_args/tests/cross/src/test/scala/tests/pc/ConvertToNamedArgumentsSuite.scala) with a single test for the scala standard lib. Should I add more tests here as well? It seems like any additional tests here would be covering the same thing as the `LspSuite` tests.

Reply:
Looks good!
You don't need to add the same test to `tests.pc.ConvertToNamedArgumentsSuite` :+1:

---

The difference between `tests.pc.ConvertToNamedArgumentsSuite` and `LspSuite` is

- While `LspSuite` checks if the action is available or not, `tests.pc....Suite` doesn't care if the action is available or not, PC test just invokes the code action anyway and sees the result.
  - So, basically PC suites are more like unit tests, and `LspSuite`s are integration tests
  - If you don't need to check the action's availability, and what you care is the behavior of the action, it should go to PC suites (as it's lighter weight)

Reply:
Got it, thanks for the clarification